CMakeLists.txt

```diff
@@ -531,36 +533,48 @@ if(LIBVNCSERVER_WITH_WEBSOCKETS)
   )
 endif(LIBVNCSERVER_WITH_WEBSOCKETS)
 
-add_library(vncclient ${LIBVNCCLIENT_SOURCES})
-add_library(vncserver ${LIBVNCSERVER_SOURCES})
 if(WIN32)
   set(ADDITIONAL_LIBS ${ADDITIONAL_LIBS} ws2_32)
   if(WITH_TIGHTVNC_FILETRANSFER)
     add_definitions(-D_WIN32_WINNT=0x0600)
   endif(WITH_TIGHTVNC_FILETRANSFER)
 endif(WIN32)
 
-target_link_libraries(vncclient
-                      ${ADDITIONAL_LIBS}
-                      ${ZLIB_LIBRARIES}
-                      ${LZO_LIBRARIES}
-                      ${JPEG_LIBRARIES}
-		      ${CRYPTO_LIBRARIES}
-                      ${GNUTLS_LIBRARIES}
-                      ${OPENSSL_LIBRARIES}
-)
-target_link_libraries(vncserver
-                      ${ADDITIONAL_LIBS}
-                      ${ZLIB_LIBRARIES}
-                      ${LZO_LIBRARIES}
-                      ${JPEG_LIBRARIES}
-		      ${PNG_LIBRARIES}
-		      ${CRYPTO_LIBRARIES}
-                      ${GNUTLS_LIBRARIES}
-                      ${OPENSSL_LIBRARIES}
-)
+set(LIBVNCSERVER_LIBRARIES )
+if(WITH_LIBVNCCLIENT)
+  add_library(vncclient ${LIBVNCCLIENT_SOURCES})
+  target_link_libraries(vncclient
+                        ${ADDITIONAL_LIBS}
+                        ${ZLIB_LIBRARIES}
+                        ${LZO_LIBRARIES}
+                        ${JPEG_LIBRARIES}
+                        ${CRYPTO_LIBRARIES}
+                        ${GNUTLS_LIBRARIES}
+                        ${OPENSSL_LIBRARIES}
```

Comment:
I moved the code a little bit around to combine both the `add_library` and `target_link_libraries` inside this `if` case.